File: fibonacci/non-recursive/Fib.java
public class Fib {
    private static void fib(int n) {
        int x1 = 1;
        int x2 = 1;
        int x3 = 0;
        System.out.println(x1);
        System.out.println(x2);

        while (x1 + x2 < n) {
            System.out.println(x1 + x2);
            x3 = x2;
            x2 += x1;
            x1 = x3;
        }

    }
    public static void main(String[] args) {
        fib(1000);
    }
}

File: fibonacci/recursive/fib.cpp
#include <iostream>

void fib(int n);
void f(int x1, int x2, int n);

int main() {
    fib(1000);
    return 0;
}

void fib(int n) {
    std::cout << 1 << std::endl;
    std::cout << 1 << std::endl;
    f(1, 1, n);
}

void f(int x1, int x2, int n) {
    if (x1 + x2 < n) {
        std::cout << x1 + x2 << std::endl;
        f(x2, x1 + x2, n);
    }
}